### ISSUE TITLE
Is the training getting slower?

### ISSUE THREAD

**Issue by Sunnydreamrain** [SeanNaren/deepspeech.pytorch#127]
Hi,

I found that the time needed to compute a batch keeps increasing. Is this right?
When a new epoch starts, the time behaviour seems to restart.

>Epoch: [11][1/48]	Time 0.470 (1.206)	Data 0.076 (0.004)	Loss 3.2168 (37.0903)	
Epoch: [11][2/48]	Time 0.407 (1.204)	Data 0.002 (0.004)	Loss 3.6915 (37.0201)	
Epoch: [11][3/48]	Time 0.393 (1.203)	Data 0.002 (0.004)	Loss 3.8567 (36.9506)	
Epoch: [11][4/48]	Time 0.403 (1.201)	Data 0.002 (0.004)	Loss 1.9861 (36.8774)	
Epoch: [11][5/48]	Time 0.538 (1.200)	Data 0.002 (0.004)	Loss 4.2020 (36.8092)	
Epoch: [11][6/48]	Time 0.538 (1.198)	Data 0.002 (0.004)	Loss 10.1045 (36.7536)	
Epoch: [11][7/48]	Time 0.663 (1.197)	Data 0.002 (0.004)	Loss 10.0306 (36.6980)	
Epoch: [11][8/48]	Time 0.664 (1.196)	Data 0.002 (0.004)	Loss 12.3729 (36.6476)	
Epoch: [11][9/48]	Time 0.663 (1.195)	Data 0.002 (0.004)	Loss 8.1055 (36.5885)	
Epoch: [11][10/48]	Time 0.806 (1.194)	Data 0.002 (0.004)	Loss 5.7821 (36.5248)	
Epoch: [11][11/48]	Time 0.818 (1.194)	Data 0.003 (0.004)	Loss 13.5125 (36.4774)	
Epoch: [11][12/48]	Time 0.817 (1.193)	Data 0.002 (0.004)	Loss 8.8798 (36.4206)	
Epoch: [11][13/48]	Time 0.815 (1.192)	Data 0.002 (0.004)	Loss 5.8867 (36.3579)	
Epoch: [11][14/48]	Time 0.943 (1.192)	Data 0.002 (0.004)	Loss 14.8824 (36.3139)	
Epoch: [11][15/48]	Time 0.929 (1.191)	Data 0.002 (0.004)	Loss 11.8922 (36.2639)	
Epoch: [11][16/48]	Time 0.935 (1.190)	Data 0.002 (0.004)	Loss 12.5397 (36.2155)	
Epoch: [11][17/48]	Time 0.931 (1.190)	Data 0.002 (0.004)	Loss 13.1328 (36.1685)	
Epoch: [11][18/48]	Time 0.938 (1.189)	Data 0.002 (0.004)	Loss 6.5284 (36.1083)	
Epoch: [11][19/48]	Time 1.051 (1.189)	Data 0.002 (0.004)	Loss 14.1728 (36.0638)	
Epoch: [11][20/48]	Time 1.068 (1.189)	Data 0.002 (0.004)	Loss 10.6078 (36.0122)	
Epoch: [11][21/48]	Time 1.066 (1.189)	Data 0.002 (0.004)	Loss 8.0026 (35.9557)	
Epoch: [11][22/48]	Time 1.067 (1.188)	Data 0.002 (0.004)	Loss 8.0095 (35.8993)	
Epoch: [11][23/48]	Time 1.059 (1.188)	Data 0.002 (0.004)	Loss 9.6850 (35.8466)	
Epoch: [11][24/48]	Time 1.061 (1.188)	Data 0.002 (0.004)	Loss 11.2493 (35.7972)	
Epoch: [11][25/48]	Time 1.257 (1.188)	Data 0.002 (0.004)	Loss 13.7587 (35.7530)	
Epoch: [11][26/48]	Time 1.259 (1.188)	Data 0.003 (0.004)	Loss 15.9801 (35.7135)	
Epoch: [11][27/48]	Time 1.255 (1.188)	Data 0.002 (0.004)	Loss 12.7662 (35.6677)	
Epoch: [11][28/48]	Time 1.256 (1.188)	Data 0.003 (0.004)	Loss 9.3659 (35.6153)	
Epoch: [11][29/48]	Time 1.255 (1.189)	Data 0.003 (0.004)	Loss 21.2329 (35.5867)	
Epoch: [11][30/48]	Time 1.253 (1.189)	Data 0.002 (0.004)	Loss 7.3173 (35.5306)	
Epoch: [11][31/48]	Time 1.394 (1.189)	Data 0.003 (0.004)	Loss 7.8257 (35.4757)	
Epoch: [11][32/48]	Time 1.399 (1.190)	Data 0.002 (0.004)	Loss 9.5787 (35.4245)	
Epoch: [11][33/48]	Time 1.384 (1.190)	Data 0.003 (0.004)	Loss 10.5516 (35.3755)	
Epoch: [11][34/48]	Time 1.498 (1.191)	Data 0.003 (0.004)	Loss 7.7515 (35.3211)	
Epoch: [11][35/48]	Time 1.494 (1.191)	Data 0.003 (0.004)	Loss 8.1408 (35.2677)	
Epoch: [11][36/48]	Time 1.477 (1.192)	Data 0.003 (0.004)	Loss 12.1592 (35.2224)	
Epoch: [11][37/48]	Time 1.616 (1.192)	Data 0.003 (0.004)	Loss 15.0813 (35.1830)	
Epoch: [11][38/48]	Time 1.619 (1.193)	Data 0.003 (0.004)	Loss 12.1004 (35.1379)	
Epoch: [11][39/48]	Time 1.612 (1.194)	Data 0.003 (0.004)	Loss 12.2281 (35.0932)	
Epoch: [11][40/48]	Time 1.792 (1.195)	Data 0.003 (0.004)	Loss 11.6189 (35.0476)	
Epoch: [11][41/48]	Time 1.787 (1.196)	Data 0.003 (0.004)	Loss 11.5063 (35.0019)	
Epoch: [11][42/48]	Time 1.840 (1.198)	Data 0.003 (0.004)	Loss 10.2395 (34.9539)	
Epoch: [11][43/48]	Time 1.935 (1.199)	Data 0.003 (0.004)	Loss 10.1281 (34.9059)	
Epoch: [11][44/48]	Time 2.067 (1.201)	Data 0.003 (0.004)	Loss 13.1505 (34.8639)	
Epoch: [11][45/48]	Time 2.171 (1.203)	Data 0.003 (0.004)	Loss 5.8545 (34.8080)	
Epoch: [11][46/48]	Time 2.379 (1.205)	Data 0.003 (0.004)	Loss 5.2639 (34.7511)	
Epoch: [11][47/48]	Time 2.782 (1.208)	Data 0.003 (0.004)	Loss 9.7186 (34.7031)	
Epoch: [11][48/48]	Time 1.336 (1.208)	Data 0.005 (0.004)	Loss 9.4000 (34.6837)	
Training Summary Epoch: [11]	Average Loss 9.897	
Validation Summary Epoch: [11]	Average WER 47.300	Average CER 24.952	
Learning rate annealed to: 0.000105
Found better validated model, saving to models/deepspeech_final.pth.tar
Epoch: [12][1/48]	Time 0.468 (1.207)	Data 0.078 (0.004)	Loss 3.3852 (34.6238)	
Epoch: [12][2/48]	Time 0.412 (1.205)	Data 0.006 (0.004)	Loss 2.0755 (34.5616)	
Epoch: [12][3/48]	Time 0.392 (1.204)	Data 0.002 (0.004)	Loss 1.7791 (34.4991)	
Epoch: [12][4/48]	Time 0.400 (1.202)	Data 0.002 (0.004)	Loss 2.1604 (34.4375)	
Epoch: [12][5/48]	Time 0.540 (1.201)	Data 0.002 (0.004)	Loss 3.4239 (34.3786)	
Epoch: [12][6/48]	Time 0.539 (1.200)	Data 0.002 (0.004)	Loss 4.8212 (34.3226)	
Epoch: [12][7/48]	Time 0.665 (1.199)	Data 0.002 (0.004)	Loss 9.9854 (34.2765)	
Epoch: [12][8/48]	Time 0.675 (1.198)	Data 0.002 (0.004)	Loss 9.2928 (34.2293)	
Epoch: [12][9/48]	Time 0.664 (1.197)	Data 0.002 (0.004)	Loss 8.6026 (34.1810)	
Epoch: [12][10/48]	Time 0.804 (1.196)	Data 0.002 (0.004)	Loss 9.2881 (34.1342)	
Epoch: [12][11/48]	Time 0.817 (1.195)	Data 0.003 (0.004)	Loss 10.8184 (34.0904)	
Epoch: [12][12/48]	Time 0.817 (1.194)	Data 0.002 (0.004)	Loss 6.0368 (34.0378)	
Epoch: [12][13/48]	Time 0.803 (1.194)	Data 0.002 (0.004)	Loss 5.6101 (33.9846)	
Epoch: [12][14/48]	Time 0.943 (1.193)	Data 0.002 (0.004)	Loss 11.7704 (33.9431)	
Epoch: [12][15/48]	Time 0.929 (1.193)	Data 0.002 (0.004)	Loss 9.6013 (33.8977)	
Epoch: [12][16/48]	Time 0.948 (1.192)	Data 0.006 (0.004)	Loss 13.7386 (33.8602)	
Epoch: [12][17/48]	Time 0.929 (1.192)	Data 0.002 (0.004)	Loss 9.9010 (33.8157)	
Epoch: [12][18/48]	Time 0.944 (1.191)	Data 0.002 (0.004)	Loss 8.2345 (33.7683)	
Epoch: [12][19/48]	Time 1.074 (1.191)	Data 0.002 (0.004)	Loss 8.2416 (33.7210)	
Epoch: [12][20/48]	Time 1.065 (1.191)	Data 0.002 (0.004)	Loss 10.8604 (33.6788)	
Epoch: [12][21/48]	Time 1.066 (1.191)	Data 0.003 (0.004)	Loss 7.9389 (33.6313)	
Epoch: [12][22/48]	Time 1.059 (1.191)	Data 0.002 (0.004)	Loss 4.9353 (33.5785)	
Epoch: [12][23/48]	Time 1.071 (1.190)	Data 0.003 (0.004)	Loss 11.4219 (33.5378)	
Epoch: [12][24/48]	Time 1.073 (1.190)	Data 0.002 (0.004)	Loss 5.7874 (33.4870)	
Epoch: [12][25/48]	Time 1.254 (1.190)	Data 0.003 (0.004)	Loss 12.1864 (33.4480)	
Epoch: [12][26/48]	Time 1.257 (1.190)	Data 0.003 (0.004)	Loss 8.8477 (33.4030)	
Epoch: [12][27/48]	Time 1.250 (1.190)	Data 0.003 (0.004)	Loss 12.0388 (33.3641)	
Epoch: [12][28/48]	Time 1.259 (1.191)	Data 0.003 (0.004)	Loss 14.2556 (33.3293)	
Epoch: [12][29/48]	Time 1.281 (1.191)	Data 0.003 (0.004)	Loss 8.7130 (33.2846)	
Epoch: [12][30/48]	Time 1.278 (1.191)	Data 0.006 (0.004)	Loss 10.5165 (33.2433)	
Epoch: [12][31/48]	Time 1.397 (1.191)	Data 0.003 (0.004)	Loss 8.0630 (33.1977)	
Epoch: [12][32/48]	Time 1.381 (1.192)	Data 0.003 (0.004)	Loss 10.2565 (33.1562)	
Epoch: [12][33/48]	Time 1.382 (1.192)	Data 0.003 (0.004)	Loss 6.5087 (33.1082)	
Epoch: [12][34/48]	Time 1.502 (1.192)	Data 0.003 (0.004)	Loss 7.5218 (33.0621)	
Epoch: [12][35/48]	Time 1.479 (1.193)	Data 0.003 (0.004)	Loss 10.9105 (33.0223)	
Epoch: [12][36/48]	Time 1.484 (1.193)	Data 0.003 (0.004)	Loss 5.1502 (32.9723)	
Epoch: [12][37/48]	Time 1.618 (1.194)	Data 0.003 (0.004)	Loss 8.5759 (32.9286)	
Epoch: [12][38/48]	Time 1.623 (1.195)	Data 0.003 (0.004)	Loss 7.8962 (32.8839)	
Epoch: [12][39/48]	Time 1.612 (1.196)	Data 0.003 (0.004)	Loss 12.1879 (32.8469)	
Epoch: [12][40/48]	Time 1.796 (1.197)	Data 0.003 (0.004)	Loss 12.8066 (32.8112)	
Epoch: [12][41/48]	Time 1.794 (1.198)	Data 0.003 (0.004)	Loss 10.0099 (32.7707)	
Epoch: [12][42/48]	Time 1.914 (1.199)	Data 0.003 (0.004)	Loss 7.4907 (32.7258)	
Epoch: [12][43/48]	Time 1.915 (1.200)	Data 0.003 (0.004)	Loss 6.3864 (32.6791)	
Epoch: [12][44/48]	Time 2.137 (1.202)	Data 0.004 (0.004)	Loss 10.0803 (32.6392)	
Epoch: [12][45/48]	Time 2.131 (1.204)	Data 0.003 (0.004)	Loss 4.8816 (32.5902)	
Epoch: [12][46/48]	Time 2.383 (1.206)	Data 0.003 (0.004)	Loss 4.3305 (32.5404)	
Epoch: [12][47/48]	Time 2.799 (1.208)	Data 0.003 (0.004)	Loss 7.1578 (32.4957)	
Epoch: [12][48/48]	Time 1.322 (1.209)	Data 0.003 (0.004)	Loss 5.8199 (32.4769)

**Comment by SiddGururani** [comment]
If you used the manifests generated by the script in the repository's data folder then this is normal behavior. The manifests are sorted by increasing length of the utterances so the number of time steps increases as training proceeds through an epoch.

**Comment by SeanNaren** [comment]
Yep as @SiddGururani this is normal behaviour since they are ordered by length :)
